### PR TITLE
refactor: use generic implementations

### PR DESCRIPTION
**Pull request by jrandolf-google**
This PR upstreams some specific implementations to generic ones. The realm methods are also now DOM-independent.

### CODE REVIEW

**Comment by OrKoN** [comment]
Tests are failing. @Lightning00Blade pls review

File: packages/puppeteer-core/src/common/Frame.ts [hunk 1]
```diff
@@ -257,7 +258,27 @@ export class CDPFrame extends Frame {
...
   ): Promise<void> {
-    return await this.isolatedRealm().setContent(html, options);
+    const {
+      waitUntil = ['load'],
+      timeout = this._frameManager.timeoutSettings.navigationTimeout(),
+    } = options;
+
+    await setPageContent(this.isolatedRealm(), html);
+
+    const watcher = new LifecycleWatcher(
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/common/Frame.ts:268]: Set up watcher before setting the page content. That way we dont miss events

File: packages/puppeteer-core/src/common/Frame.ts [hunk 2]
```diff
@@ -257,7 +258,27 @@ export class CDPFrame extends Frame {
       waitUntil?: PuppeteerLifeCycleEvent | PuppeteerLifeCycleEvent[];
     } = {}
   ): Promise<void> {
-    return await this.isolatedRealm().setContent(html, options);
+    const {
+      waitUntil = ['load'],
+      timeout = this._frameManager.timeoutSettings.navigationTimeout(),
+    } = options;
+
+    await setPageContent(this.isolatedRealm(), html);
```
Comment (Lightning00Blade) [inline packages/puppeteer-core/src/common/Frame.ts:266]: Can this now be moved inside a common implementation?

File: packages/puppeteer-core/src/common/Frame.ts (hunk 1 above)
Comment (OrKoN) [reply packages/puppeteer-core/src/common/Frame.ts:268]: I think the previous version was exactly the same

File: packages/puppeteer-core/src/common/Frame.ts (hunk 2 above)
Comment (OrKoN) [reply packages/puppeteer-core/src/common/Frame.ts:266]: Looks like we can now

**Review by OrKoN — APPROVED** [review]

File: packages/puppeteer-core/src/common/Frame.ts (hunk 1 above)
Comment (Lightning00Blade) [reply packages/puppeteer-core/src/common/Frame.ts:268]: I don't see how that is relevant, when we should improve it.

**Pull request merged**